### PR TITLE
Add a docker template

### PR DESCRIPTION
**Pull request by zabil**
Add a docker file to this template so that users can use it as reference when they do a `gauge init js`

Signed-off-by: Zabil Cheriya Maliackal <zabilcm@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Building the image
+# 	docker build -t gauge-taiko .
+# Running the image
+# 	docker run  --rm -it -v ${PWD}/reports:/gauge/reports gauge-taiko
+
+# This image uses the official node base image.
+FROM node
+ 
+# The Taiko installation downloads and installs the chromium required to run the tests. 
+# However, we need the chromium dependencies installed in the environment. These days, most # Dockerfiles just install chrome to get the dependencies.
```
Comment (sriv) [inline Dockerfile]:
```suggestion
# However, we need the chromium dependencies installed in the environment. These days, most 
# Dockerfiles just install chrome to get the dependencies.
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# installed without root privileges
+ENV NPM_CONFIG_PREFIX=/home/gauge/.npm-packages
+ENV PATH="${NPM_CONFIG_PREFIX}/bin:${PATH}"
+
+# Add the Taiko browser arguments
+ENV TAIKO_BROWSER_ARGS=--no-sandbox,--start-maximized,--disable-dev-shm-usage
+ENV headless_chrome=true
+ENV TAIKO_SKIP_DOCUMENTATION=true
+#ENV TAIKO_SKIP_CHROMIUM_DOWNLOAD=true
+#ENV TAIKO_BROWSER_PATH=/usr/bin/google-chrome
```
Comment (sriv) [inline Dockerfile:31]: can remove these?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ 
+# Add test code
+ADD . /gauge
+
+# Set working directory
+WORKDIR /gauge
+ 
+# Copy the local working folder
+COPY . .
```
Comment (sriv) [inline Dockerfile:37]: Do we need both [`ADD`](https://github.com/getgauge/template-js/pull/2/files#diff-3254677a7917c6c01f55212f86c57fbfR31) and `COPY`?

**Review by sriv — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (zabil) [reply Dockerfile:31]: I kept it there in case they want to use the bundled chrome. I will add a comment.

**Review by sriv — APPROVED** [review]

**Pull request merged**